### DISCUSSION TITLE
linker error undefined reference to `Xsns113(unsigned int)'

### DISCUSSION THREAD

**Discussion by gneiss15** [arendst/Tasmota#19959] [Support request] [answered]
I try to add my own sensor.
I use the next availbale number/ID (as far as I can see it's the 113) and create a file xsns_113.ino inside dir tasmota_xsns_sensor.
The file compiles fine (checked).

The file contains the Interface-function:
bool Xsns113( uint32_t function ) { ... }
as well as the definition
#define XSNS_113	113

At the end of the build (platformio & VSCode) I got the "undefined reference" error.

Can anybody explain me where to search ?
I could not figure out why this symbol was excluded from link..

It's listed inside "firmware.map" at two places:
```
_Z7Xsns113j                                       /tmp/cc6nT645.ltrans9.ltrans.o
                                                  .pio/build/tasmota32-DE/src/tasmota.ino.cpp.o (symbol from plugin)

_ZN4XSML7Xsns113Ej                                .pio/build/tasmota32-DE/src/tasmota.ino.cpp.o (symbol from plugin)
```

**Comment by barbudor** [comment]
Hard to tell without more details including a complete build log

**Reply by gneiss15** [reply]
I am still "fighting" with this IDE (VSCode + platformio + Tasmota)...
I had used other IDEs a long time, but still know to less here to supply a build log.
Where can I find it ?

There is msg at start of build 
Verbose mode can be enabled via `-v, --verbose` option
Doing this, by modifying "add_c_flags.py" in Tasmota/pio-tools will recompile everything..
So how add this flag only for the projekt ?
I had added the content of the Terminal of a "normal" build below...

BTW.: "fighting" with this IDE...
If there is a compile time problem inside the console the error is "underlined". Cliking the does nothing. Switching to tab "problems" I got "file not found".
I guess that is because I use a portable version of VSCode and the file was searched with the "data" dir as root.
The portable home is /home/gneiss/VSCode-linux-x64/data.
I now add:
mkdir -p /home/gneiss/VSCode-linux-x64/data/Tasmota/home/gneiss/VSCode-linux-x64
ln -s /home/gneiss/VSCode-linux-x64/data /home/gneiss/VSCode-linux-x64/data/Tasmota/home/gneiss/VSCode-linux-x64/data
Now the error-source file will be opened (but in an extra editor, because VSCode seams to thread this symlinked file as differnt from the original one).

` *  Task wird ausgeführt: platformio run --environment tasmota32-DE 

Processing tasmota32-DE (framework: arduino; platform: https://github.com/tasmota/platform-espressif32/releases/download/2023.10.03/platform-espressif32.zip; board: esp32)
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
-DCONFIG_TASMOTA_FLASHMODE_DIO
*** use provided user_config_override.h as planned ***
*** use provided platformio_override.ini as planned ***
*** use provided platformio_tasmota_cenv.ini as planned ***
*** use provided platformio_tasmota_core3_env.ini as planned ***
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp32.html
PLATFORM: Espressif 32 (2023.10.3) > Espressif Generic ESP32 >= 4M Flash PSRAM, Tasmota 2880k Code/OTA, 320k FS
HARDWARE: ESP32 160MHz, 320KB RAM, 4MB Flash
DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES: 
 - framework-arduinoespressif32 @ 2.0.14 
 - tool-esptoolpy @ 1.40700.0+dev (4.7.0) 
 - tool-mklittlefs @ 3.2.0 
 - tool-mkspiffs @ 2.230.0 (2.30) 
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
Converting tasmota.ino
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ strict
Found 132 compatible libraries
Scanning dependencies...
Dependency Graph
|-- Adafruit CCS811 Library @ 1.0.0
|-- Adafruit LittlevGL Glue Library @ 2.0.0
|-- Adafruit MAX31865 library @ 1.0.1
|-- Adafruit MCP9808 Library @ 1.1.2
|-- Adafruit PM25 AQI Sensor @ 1.0.6
|-- Adafruit SGP30 Sensor @ 1.2.0
|-- Adafruit SGP40 Sensor @ 1.1.0
|-- Adafruit VEML7700 Library @ 1.0.0
|-- Adafruit seesaw Library @ 1.3.1
|-- ESP8266Audio @ 1.9.5
|-- BME8563
|-- EEPROM 24C128_256_521
|-- FFat @ 2.0.0
|-- FS @ 2.0.0
|-- HttpClient light @ 1.0
|-- I2Cdevlib-Core
|-- LibTeleinfo @ 1.1.5
|-- MPU6050 I2C
|-- Mutichannel_Gas_Sensor @ 0.0.1
|-- NTP Library
|-- NimBLE-Arduino @ 1.4.1
|-- OpenTherm Library @ 0.9.0
|-- Process_control @ 1.0.0
|-- ESP Mail Client @ 3.4.9
|-- ScioSense ENS16x @ 8.0.0
|-- ScioSense ENS210 @ 3.0.0
|-- Sensirion I2C SGP41 @ 0.1.0
|-- SevenSegmentTM1637 @ 1.1.0
|-- UdpListener @ 1.0
|-- Unishox Compressor Decompressor highly customized and optimized for ESP8266 and Tasmota @ 1.0
|-- VL53L0X @ 1.0.2
|-- VL53L1X @ 1.0.1
|-- TLS mini client derived from Arduino @ 1.0
|-- Zip-readonly-FS @ 1.0
|-- Berry Tasmota mapping @ 1.0
|-- Berry @ 7.0
|-- Berry int64 implementation for 32 bits architecture @ 1.0
|-- Berry mapping to C @ 1.0
|-- Berry Matter protocol implementation @ 0.1
|-- headers
|-- ccronexpr
|-- WebServer @ 2.0.0
|-- ESP32-to-ESP8266-compat @ 0.0.2
|-- ams @ 1.0
|-- Berry mapping to LVGL @ 1.0
|-- lvgl @ 8.3.10
|-- arduino-mcp2515-1.0.1
|-- re1.5 @ 0.9
|-- LVGL_assets @ 7.0
|-- A4988_Stepper @ 0.0.1
|-- Adafruit Fingerprint Sensor Library @ 2.1.0
|-- Adafruit GFX Library @ 1.5.6
|-- Adafruit LED Backpack Library @ 1.1.6
|-- Adafruit SH1106-gemu-1.0 @ 1.0
|-- Adafruit SSD1306 @ 1.3.0
|-- Adafruit SSD1331 OLED Driver Library for Arduino @ 1.2.0
|-- Adafruit TSL2591
|-- Arduino ST7789 Library @ 0.9.5
|-- Micro-RTSP @ 0.1.6
|-- DHT sensor library @ 1.4.3
|-- DNSServer @ 2.0.0
|-- DS2480 Library @ 0.1.0
|-- ESP8266SAM @ 1.0.1
|-- Ethernet @ 2.0.0
|-- FT5206_Library @ 1.0.0
|-- FrogmoreScd30
|-- FrogmoreScd40
|-- GT911
|-- ILI9341 @ 1.0.0
|-- IRremoteESP8266 @ 2.8.6
|-- JSMN JSON parser customized and optimized for ESP8266 and Tasmota @ 1.0
|-- KeeloqLib @ 1.1
|-- TasmotaLList @ 1.0
|-- LOLIN_HP303B @ 1.0.0
|-- LedControl @ 1.0.6
|-- LinkedList
|-- LiquidCrystal_I2C
|-- LittleFS @ 2.0.0
|-- MFRC522 @ 1.4.7
|-- MLX90640
|-- MPU_accel
|-- NeoPixelBus @ 2.6.7
|-- NewPing @ 1.9.1
|-- OneWire @ 2.3.2
|-- PubSubClient @ 2.8
|-- RA8876 @ 1.0.2
|-- rc-switch @ 1.0.0
|-- RF24 @ 1.3.3
|-- Ext-printf @ 1.0
|-- SD @ 2.0.0
|-- SD_MMC @ 2.0.0
|-- SPI @ 2.0.0
|-- SSD3115 @ 1.0
|-- Sensirion I2C SEN5X @ 0.2.0
|-- TM1638plus @ 1.7.0
|-- TasmotaModbus @ 3.6.0
|-- TasmotaSerial @ 3.6.0
|-- Ticker @ 2.0.0
|-- Joba_Tsl2561 @ 2.0.10
|-- WiFiClientSecure @ 2.0.0
|-- WiFi @ 2.0.0
|-- Wire @ 2.0.0
|-- XPT2046_Touchscreen
|-- TTGO TWatch Library @ 1.2.0
|-- base64 @ 1.1.1
|-- BME68x
|-- Waveshare esp 2.9 inch e-paper display driver @ 1.0
|-- Waveshare esp 4.2 inch e-paper display driver @ 1.0
|-- Display renderer @ 1.0
|-- ES7210 @ 1.0.0
|-- ES7243e @ 1.0.0
|-- ES8156 @ 1.0.0
|-- ES8311 @ 1.0.0
|-- ESP KNX IP Library @ 0.5.2
|-- HPMA115S0 Arduino Library @ 1.0.0
|-- mp3_shine_esp32 @ 1.0.0
|-- BearSSL @ 0.6
|-- universal display Library @ 0.1
|-- WM8960 @ 1.0.0
|-- ESP32-HomeKit @ 1.0.0
Building in release mode
Compiling .pio/build/tasmota32-DE/src/tasmota.ino.cpp.o
Retrieved `.pio/build/tasmota32-DE/src/tasmota_support/homekit.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/src/tasmota_xsns_sensor/xsns_62_esp32_mi_homekit.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib59e/Wire/Wire.cpp.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib440/Adafruit_CCS811-1.0.0.14/Adafruit_CCS811.cpp.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib02f/FS/FS.cpp.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib02f/FS/vfs_api.cpp.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib4f0/Ticker/Ticker.cpp.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib749/freetype/base/ftbase.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib749/freetype/base/ftbdf.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib749/freetype/base/ftbbox.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib749/freetype/base/ftbitmap.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib4f0/libTicker.a' from cache
Retrieved `.pio/build/tasmota32-DE/lib02f/libFS.a' from cache
Retrieved `.pio/build/tasmota32-DE/lib749/freetype/base/ftdebug.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib59e/libWire.a' from cache
Retrieved `.pio/build/tasmota32-DE/lib440/libAdafruit_CCS811-1.0.0.14.a' from cache
Retrieved `.pio/build/tasmota32-DE/lib749/freetype/base/ftinit.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib749/freetype/bdf/bdf.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib749/freetype/base/ftsystem.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib749/freetype/base/ftglyph.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib749/freetype/cache/ftcache.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib749/freetype/cff/cff.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib749/freetype/sfnt/sfnt.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib749/freetype/smooth/smooth.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib749/freetype/truetype/truetype.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/core/lv_disp.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/core/lv_event.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/core/lv_group.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/core/lv_indev.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/core/lv_indev_scroll.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/core/lv_obj.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/core/lv_obj_class.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/core/lv_obj_draw.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/core/lv_obj_pos.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/core/lv_obj_scroll.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib749/libfreetype.a' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/core/lv_obj_style.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/core/lv_obj_tree.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/core/lv_refr.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/core/lv_obj_style_gen.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/core/lv_theme.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/arm2d/lv_gpu_arm2d.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/lv_draw.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/lv_draw_arc.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/lv_draw_img.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/lv_draw_layer.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/lv_draw_line.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/lv_draw_label.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/lv_draw_rect.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/lv_draw_mask.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/lv_draw_transform.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/lv_draw_triangle.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/lv_img_decoder.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/lv_img_cache.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/lv_img_buf.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/nxp/pxp/lv_draw_pxp.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/nxp/pxp/lv_draw_pxp_blend.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/nxp/pxp/lv_gpu_nxp_pxp.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/nxp/pxp/lv_gpu_nxp_pxp_osa.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/nxp/vglite/lv_draw_vglite.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/nxp/vglite/lv_draw_vglite_arc.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/nxp/vglite/lv_draw_vglite_blend.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/nxp/vglite/lv_draw_vglite_line.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/nxp/vglite/lv_draw_vglite_rect.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/nxp/vglite/lv_vglite_buf.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/nxp/vglite/lv_vglite_utils.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/renesas/lv_gpu_d2_draw_label.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/renesas/lv_gpu_d2_ra6m3.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/sdl/lv_draw_sdl_bg.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/sdl/lv_draw_sdl_composite.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/sdl/lv_draw_sdl_img.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/sdl/lv_draw_sdl_label.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/sdl/lv_draw_sdl_layer.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/sdl/lv_draw_sdl_line.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/sdl/lv_draw_sdl_mask.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/sdl/lv_draw_sdl_polygon.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/sdl/lv_draw_sdl_rect.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/sdl/lv_draw_sdl_stack_blur.c.o' from cache
Retrieved `.pio/build/tasmota32-DE/lib193/lvgl/draw/sdl/lv_draw_sdl_texture_cache.c.o' from cache
Archiving .pio/build/tasmota32-DE/lib193/liblvgl.a
Indexing .pio/build/tasmota32-DE/lib193/liblvgl.a
Linking .pio/build/tasmota32-DE/firmware.elf
/home/gneiss/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld: /tmp/ccFI7sEE.ltrans9.ltrans.o:(.rodata+0x8b4): undefined reference to `Xsns113(unsigned int)'
collect2: error: ld returned 1 exit status
*** [.pio/build/tasmota32-DE/firmware.elf] Error 1
==================================================================================== [FAILED] Took 63.94 seconds ====================================================================================

Environment    Status    Duration
-------------  --------  ------------
tasmota32-DE   FAILED    00:01:03.939
=============================================================================== 1 failed, 0 succeeded in 00:01:03.939 ===============================================================================

 *  Der Terminalprozess "platformio 'run', '--environment', 'tasmota32-DE'" wurde mit folgendem Exitcode beendet: 1. 
 *  Das Terminal wird von Aufgaben wiederverwendet, drücken Sie zum Schließen eine beliebige Taste. 
`

**Reply by barbudor** [reply]
If you find that symbol in the map, your code seems to be compiled
But how could you get a map if you link doesn't suceed ...

Does the same problem happens when compiling tasmota32 (and not tasmoa32-DE) ?


I found your fork in your github but not those changes
Could you push your latest version on your github so I could try it ? (but not before tonight unfortunately, got to get back to work now)

**Reply by arendst** [reply]
BTW XSNS_113 is already used for driver ``xsns_113_hc8.ino``

**Reply by gneiss15** [reply]
Thanks for You interrest...

> But how could you get a map if you link doesn't suceed ...

Good question. It seams that the linker generates that file even when he missed a symbol..

> Does the same problem happens when compiling tasmota32 (and not tasmoa32-DE) ?

Yes.

>  found your fork in your github...

I will upload my changes.. But keep in mind that my additional sensor is just at a verry verry beta state (no real function right now).
I just try if my serial classes are working (by making a loop hw -> sw serial)

Some more informations.
In the mean while I had done some more investigations/tests. 

1. compile the code by using 
pio run -v -j1 2>&1 | tee /home/gneiss/VSCode-linux-x64/data/Tasmota/build_output/build.log
This will give me a full build.log...
-> But this doesn't realy help me :-(

2. Use a (post_extra_)script that
- logs the env and projenv to files (in same dir as above)
- uses: projenv.Append(CXXFLAGS=["-v"]) to make the normal build verbose
-> But this doesn't realy help me, too :-(

3. Examine the map file in more detail
All functions "bool Xsns...(uint32_t ...)" (except mine) are listed inside a ".text" section that comes from a file named:
"/tmp/<temp_file_name>.ltrans`<n>`.ltrans.o".
Where <temp_file_name> changes from compilation to compilation and `<n>` is 9..11

Exspecially (my Xsns113 is derived from Xsns53) when comparing those two:
```
# The symbol was fond inside these files
_Z6Xsns53j                                        /tmp/ccFxh6cX.ltrans10.ltrans.o
                                                  .pio/build/tasmota32/src/tasmota.ino.cpp.o (symbol from plugin)
                                                  /tmp/ccFxh6cX.ltrans9.ltrans.o

# This o-file contains the code
 .text          0x0000000040111eb8     0x6a37 /tmp/ccFxh6cX.ltrans10.ltrans.o
...
                0x0000000040117cc4                _Z6Xsns53j

_Z7Xsns113j                                       /tmp/ccFxh6cX.ltrans9.ltrans.o
                                                  .pio/build/tasmota32/src/tasmota.ino.cpp.o (symbol from plugin)
```
/tmp/ccFxh6cX.ltrans9.ltrans.o seams to contain tasmota_xx2c_global/xsns_interface.ino, because all Xsns<No> are listed here.
So it seams that (for whatever reason) Xsns113 was not "packed" into one of these "ltrans".

Now I have to find out how this "packing" was done.
There is a hint during build, that ...ino was converted to ...cpp...
But till now I didn't find how this is done.

**Reply by gneiss15** [reply]
Repro synced with orig & commited my changes

**Reply by gneiss15** [reply]
My sensor is now 114..

**Reply by gneiss15** [reply]
Ok, just to clearify. After syncing and movin my code to Xsns114 the same problem exists. Now Xsns114 was not found

**Reply by barbudor** [reply]
trying it now

**Reply by barbudor** [reply]
Stupid me, there was no `#define USE_SML_X`, so your code was ot used

**Reply by barbudor** [reply, marked as answer]
Ok, I found your problem: it's related to your using a namespace
Tasmota is expecting to call `Xsns114()`, not `XSML::Xsns114()`

Moving `Xsns114()` out of the namespce, plus adding some `XSML::` where needed and it's solved

See my PR : https://github.com/gneiss15/Tasmota/pull/1

**Reply by gneiss15** [reply]
OMG,
the namespace is something that I had added just before trying to test the serial comunication...
And I had already written the problem/Answer in my first post :-(
There are two entries for 'Xsns113' inside the map:
- _Z7Xsns113j
  That's usage of Xsns113 inside xsns_interface.ino
- _ZN4XSML7Xsns113Ej
  That's the definition of XSML::Xsns113 inside my code

As usual. you (I) should always think twice ;-)

Of course the namespace should be used as You suggest in Your PR...

Thank you very mutch for Your support.

**Comment by gneiss15** [comment]
@arendst
> BTW XSNS_113 is already used for driver `xsns_113_hc8.ino`

Not in my current fork... OK, no problem I will use another number :-)
